### PR TITLE
Setup: add express template code

### PR DESCRIPTION
**Pull request by avinash360**
- add express code

- update package.json

### CODE REVIEW

**Comment by avinash360** [comment]
@nimish-gupta this pull request depends upon https://github.com/pesto-students/batch-9-Posto/pull/30

**Comment by avinash360** [comment]
@nimish-gupta this code has been rebased with https://github.com/pesto-students/batch-9-Posto/pull/30

**Comment by nimish-gupta** [comment]
> @nimish-gupta this code has been rebased with #30

why?

If yes then close that PR, and then change the `var` in this PR.

**Comment by avinash360** [comment]
@nimish-gupta in that pr there were merge conflicts, so i created a new branch and after rebasing it properly i have sent this pull request.

var has been changed to const in another pull request in which we solved eslint issues.

File: server/.babelrc [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+    "presets": [
+        "@babel/preset-env"
+    ]
+}
```
Comment (rishabh1403) [inline server/.babelrc]: new line

File: server/.env [hunk 2]
```diff
@@ -0,0 +1 @@
+MY_SECRET=somesupersecretpasswword
```
Comment (rishabh1403) [inline server/.env]: new line

File: server/app.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+var createError = require('http-errors');
+var express = require('express');
```
Comment (rishabh1403) [inline server/app.js]: const instead of var

File: server/app.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+var logger = require('morgan');
+
+var indexRouter = require('./routes/index');
+var usersRouter = require('./routes/users');
+
+var app = express();
+
+// view engine setup
+app.set('views', path.join(__dirname, 'views'));
+app.set('view engine', 'jade');
```
Comment (rishabh1403) [inline server/app.js]: why `jade` ? Are you planning server side rendering?

File: server/app.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+app.use(logger('dev'));
+app.use(express.json());
+app.use(express.urlencoded({ extended: false }));
+app.use(cookieParser());
+app.use(express.static(path.join(__dirname, 'public')));
```
Comment (rishabh1403) [inline server/app.js]: are you planning to serve react from backend server itself?

File: server/app.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+app.use('/', indexRouter);
+app.use('/users', usersRouter);
+
+// catch 404 and forward to error handler
+app.use(function (req, res, next) {
+  next(createError(404));
```
Comment (rishabh1403) [inline server/app.js]: where is `createError` defined ?

File: server/app.js (hunk 3 above)
Comment (avinash360) [reply server/app.js]: This has been done in next pull request

File: server/app.js (hunk 4 above)
Comment (avinash360) [reply server/app.js]: no

File: server/app.js (hunk 5 above)
Comment (avinash360) [reply server/app.js]: no

File: server/app.js (hunk 6 above)
Comment (avinash360) [reply server/app.js]: at the top of this file, we have the following line

File: server/app.js (hunk 3 above)
Comment (rishabh1403) [reply server/app.js]: But we can't really merge this into master, this is not clean, either rebase from another pr or clean it :)

File: server/.babelrc (hunk 1 above)
Comment (avinash360) [reply server/.babelrc]: done

File: server/.env (hunk 2 above)
Comment (avinash360) [reply server/.env]: done

File: server/app.js (hunk 3 above)
Comment (avinash360) [reply server/app.js]: done

File: server/app.js (hunk 4 above)
Comment (avinash360) [reply server/app.js]: removed

File: server/app.js (hunk 5 above)
Comment (avinash360) [reply server/app.js]: removed

**Pull request merged**